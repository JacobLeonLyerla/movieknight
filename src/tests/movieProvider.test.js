import React from "react";
import ReactDOM from "react-dom";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import MovieProvider from "../contexts/movieProvider";

Enzyme.configure({ adapter: new Adapter() });

describe("<MovieProvider />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MovieProvider />, div);
  });
  it("check snapshot", () => {
    let element = shallow(<MovieProvider />);
    expect(element).toMatchSnapshot();
  });
  it("correct number of children", () => {
    const wrapper = shallow(<MovieProvider />);
    expect(wrapper).toHaveLength(1);
  });
it("state set as expected",()=>{
  const wrapper = shallow(<MovieProvider/>)
  expect(typeof wrapper.state("page")).toBe('number')
  expect(typeof wrapper.state("totalPages")).toBe('number')
  expect(typeof wrapper.state("movies")).toBe('object')
})

 it("fetchMovies is a function",()=>{
   const wrapper = shallow(<MovieProvider/>)
   wrapper.instance().fetchMovies()
 })
 
 it("setPage is a function",()=>{
  const wrapper = shallow(<MovieProvider/>)
  wrapper.instance().setPage()
})
});
