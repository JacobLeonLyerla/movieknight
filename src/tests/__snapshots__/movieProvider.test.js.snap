// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieProvider /> check snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieProvider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "value": Object {
        "actions": Object {
          "fetchMovies": [Function],
          "handleSearch": [Function],
          "handleType": [Function],
          "setPage": [Function],
        },
        "movieData": Object {
          "movies": Array [],
          "page": 1,
          "search": "",
          "totalPages": 0,
          "type": "now_playing",
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": Object {},
        "_currentRenderer2": null,
        "_currentValue": Object {},
        "_currentValue2": Object {},
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "value": Object {
          "actions": Object {
            "fetchMovies": [Function],
            "handleSearch": [Function],
            "handleType": [Function],
            "setPage": [Function],
          },
          "movieData": Object {
            "movies": Array [],
            "page": 1,
            "search": "",
            "totalPages": 0,
            "type": "now_playing",
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": Object {},
          "_currentRenderer2": null,
          "_currentValue": Object {},
          "_currentValue2": Object {},
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
